### ISSUE TITLE
v2.1.3: Unit test errors

### ISSUE THREAD

**Issue by spatanjali** [SAP/olingo-jpa-processor-v4#347]
@wog48  Unable to build the v2.1.3 release branch due to the following seven unit test errors within the `odata-jpa-processor` module

```
[ERROR]   TestScalarDbFunctions.testFilterOnFunction:68 {"error":{"code":null,"message":"Exception [EclipseLink-4002] (Eclipse Persistence Services - 4.0.3.v202405220658): org.eclipse.persistence.exceptions.DatabaseException\r\nInternal Exception: java.sql.SQLSyntaxErrorException: user lacks privilege or object not found: PopulationDensity in statement [SELECT E0.\"CodePublisher\" S0, E0.\"CountryISOCode\" S1, E0.\"Area\" S2, E0.\"Population\" S3, E0.\"DivisionCode\" S4, E0.\"ParentDivisionCode\" S5, E0.\"ParentCodeID\" S6, E0.\"AlternativeCode\" S7, E0.\"CodeID\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") > ?)]\r\nError Code: -5501\r\nCall: SELECT E0.\"CodePublisher\" S0, E0.\"CountryISOCode\" S1, E0.\"Area\" S2, E0.\"Population\" S3, E0.\"DivisionCode\" S4, E0.\"ParentDivisionCode\" S5, E0.\"ParentCodeID\" S6, E0.\"AlternativeCode\" S7, E0.\"CodeID\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") > ?)\r\n\tbind => [1]\r\nQuery: DataReadQuery(sql=\"SELECT E0.\"CodePublisher\" S0, E0.\"CountryISOCode\" S1, E0.\"Area\" S2, E0.\"Population\" S3, E0.\"DivisionCode\" S4, E0.\"ParentDivisionCode\" S5, E0.\"ParentCodeID\" S6, E0.\"AlternativeCode\" S7, E0.\"CodeID\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") > ?)\")"}} ==> expected: <200> but was: <500>

[ERROR]   TestScalarDbFunctions.testFilterOnFunctionAndMultiply:96 {"error":{"code":null,"message":"Exception [EclipseLink-4002] (Eclipse Persistence Services - 4.0.3.v202405220658): org.eclipse.persistence.exceptions.DatabaseException\r\nInternal Exception: java.sql.SQLSyntaxErrorException: user lacks privilege or object not found: PopulationDensity in statement [SELECT E0.\"CodeID\" S0, E0.\"DivisionCode\" S1, E0.\"ParentCodeID\" S2, E0.\"Area\" S3, E0.\"CodePublisher\" S4, E0.\"Population\" S5, E0.\"CountryISOCode\" S6, E0.\"AlternativeCode\" S7, E0.\"ParentDivisionCode\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE ((\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") * ?) > ?)]\r\nError Code: -5501\r\nCall: SELECT E0.\"CodeID\" S0, E0.\"DivisionCode\" S1, E0.\"ParentCodeID\" S2, E0.\"Area\" S3, E0.\"CodePublisher\" S4, E0.\"Population\" S5, E0.\"CountryISOCode\" S6, E0.\"AlternativeCode\" S7, E0.\"ParentDivisionCode\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE ((\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") * ?) > ?)\r\n\tbind => [1000000.0, 100]\r\nQuery: DataReadQuery(sql=\"SELECT E0.\"CodeID\" S0, E0.\"DivisionCode\" S1, E0.\"ParentCodeID\" S2, E0.\"Area\" S3, E0.\"CodePublisher\" S4, E0.\"Population\" S5, E0.\"CountryISOCode\" S6, E0.\"AlternativeCode\" S7, E0.\"ParentDivisionCode\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE ((\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") * ?) > ?)\")"}} ==> expected: <200> but was: <500>

[ERROR]   TestScalarDbFunctions.testFilterOnFunctionAndMultiply:96 {"error":{"code":null,"message":"Exception [EclipseLink-4002] (Eclipse Persistence Services - 4.0.3.v202405220658): org.eclipse.persistence.exceptions.DatabaseException\r\nInternal Exception: java.sql.SQLSyntaxErrorException: user lacks privilege or object not found: PopulationDensity in statement [SELECT E0.\"CodeID\" S0, E0.\"CountryISOCode\" S1, E0.\"CodePublisher\" S2, E0.\"ParentCodeID\" S3, E0.\"DivisionCode\" S4, E0.\"AlternativeCode\" S5, E0.\"Area\" S6, E0.\"ParentDivisionCode\" S7, E0.\"Population\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE ((\"OLINGO\".\"PopulationDensity\"(?, E0.\"Population\") * ?) > ?)]\r\nError Code: -5501\r\nCall: SELECT E0.\"CodeID\" S0, E0.\"CountryISOCode\" S1, E0.\"CodePublisher\" S2, E0.\"ParentCodeID\" S3, E0.\"DivisionCode\" S4, E0.\"AlternativeCode\" S5, E0.\"Area\" S6, E0.\"ParentDivisionCode\" S7, E0.\"Population\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE ((\"OLINGO\".\"PopulationDensity\"(?, E0.\"Population\") * ?) > ?)\r\n\tbind => [13079087, 1000000.0, 1000]\r\nQuery: DataReadQuery(sql=\"SELECT E0.\"CodeID\" S0, E0.\"CountryISOCode\" S1, E0.\"CodePublisher\" S2, E0.\"ParentCodeID\" S3, E0.\"DivisionCode\" S4, E0.\"AlternativeCode\" S5, E0.\"Area\" S6, E0.\"ParentDivisionCode\" S7, E0.\"Population\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE ((\"OLINGO\".\"PopulationDensity\"(?, E0.\"Population\") * ?) > ?)\")"}} ==> expected: <200> but was: <500>

[ERROR]   TestScalarDbFunctions.testFilterOnFunctionAndMultiply:96 {"error":{"code":null,"message":"Exception [EclipseLink-4002] (Eclipse Persistence Services - 4.0.3.v202405220658): org.eclipse.persistence.exceptions.DatabaseException\r\nInternal Exception: java.sql.SQLSyntaxErrorException: user lacks privilege or object not found: PopulationDensity in statement [SELECT E0.\"CountryISOCode\" S0, E0.\"ParentDivisionCode\" S1, E0.\"CodeID\" S2, E0.\"Population\" S3, E0.\"DivisionCode\" S4, E0.\"AlternativeCode\" S5, E0.\"ParentCodeID\" S6, E0.\"Area\" S7, E0.\"CodePublisher\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (\"OLINGO\".\"PopulationDensity\"((E0.\"Area\" / ?), E0.\"Population\") > ?)]\r\nError Code: -5501\r\nCall: SELECT E0.\"CountryISOCode\" S0, E0.\"ParentDivisionCode\" S1, E0.\"CodeID\" S2, E0.\"Population\" S3, E0.\"DivisionCode\" S4, E0.\"AlternativeCode\" S5, E0.\"ParentCodeID\" S6, E0.\"Area\" S7, E0.\"CodePublisher\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (\"OLINGO\".\"PopulationDensity\"((E0.\"Area\" / ?), E0.\"Population\") > ?)\r\n\tbind => [1000000, 1000]\r\nQuery: DataReadQuery(sql=\"SELECT E0.\"CountryISOCode\" S0, E0.\"ParentDivisionCode\" S1, E0.\"CodeID\" S2, E0.\"Population\" S3, E0.\"DivisionCode\" S4, E0.\"AlternativeCode\" S5, E0.\"ParentCodeID\" S6, E0.\"Area\" S7, E0.\"CodePublisher\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (\"OLINGO\".\"PopulationDensity\"((E0.\"Area\" / ?), E0.\"Population\") > ?)\")"}} ==> expected: <200> but was: <500>

[ERROR]   TestScalarDbFunctions.testFilterOnFunctionAndMultiply:96 {"error":{"code":null,"message":"Exception [EclipseLink-4002] (Eclipse Persistence Services - 4.0.3.v202405220658): org.eclipse.persistence.exceptions.DatabaseException\r\nInternal Exception: java.sql.SQLSyntaxErrorException: user lacks privilege or object not found: PopulationDensity in statement [SELECT E0.\"ParentCodeID\" S0, E0.\"Area\" S1, E0.\"DivisionCode\" S2, E0.\"AlternativeCode\" S3, E0.\"CodePublisher\" S4, E0.\"Population\" S5, E0.\"CountryISOCode\" S6, E0.\"ParentDivisionCode\" S7, E0.\"CodeID\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE ((\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") * ?) > ?)]\r\nError Code: -5501\r\nCall: SELECT E0.\"ParentCodeID\" S0, E0.\"Area\" S1, E0.\"DivisionCode\" S2, E0.\"AlternativeCode\" S3, E0.\"CodePublisher\" S4, E0.\"Population\" S5, E0.\"CountryISOCode\" S6, E0.\"ParentDivisionCode\" S7, E0.\"CodeID\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE ((\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") * ?) > ?)\r\n\tbind => [1000000.0, 1000]\r\nQuery: DataReadQuery(sql=\"SELECT E0.\"ParentCodeID\" S0, E0.\"Area\" S1, E0.\"DivisionCode\" S2, E0.\"AlternativeCode\" S3, E0.\"CodePublisher\" S4, E0.\"Population\" S5, E0.\"CountryISOCode\" S6, E0.\"ParentDivisionCode\" S7, E0.\"CodeID\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE ((\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") * ?) > ?)\")"}} ==> expected: <200> but was: <500>

[ERROR]   TestScalarDbFunctions.testFilterOnFunctionAndProperty:56 {"error":{"code":null,"message":"Exception [EclipseLink-4002] (Eclipse Persistence Services - 4.0.3.v202405220658): org.eclipse.persistence.exceptions.DatabaseException\r\nInternal Exception: java.sql.SQLSyntaxErrorException: user lacks privilege or object not found: PopulationDensity in statement [SELECT E0.\"ParentDivisionCode\" S0, E0.\"DivisionCode\" S1, E0.\"ParentCodeID\" S2, E0.\"Population\" S3, E0.\"AlternativeCode\" S4, E0.\"Area\" S5, E0.\"CodePublisher\" S6, E0.\"CountryISOCode\" S7, E0.\"CodeID\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (((\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") * ?) > ?) AND (E0.\"ParentDivisionCode\" = ?))]\r\nError Code: -5501\r\nCall: SELECT E0.\"ParentDivisionCode\" S0, E0.\"DivisionCode\" S1, E0.\"ParentCodeID\" S2, E0.\"Population\" S3, E0.\"AlternativeCode\" S4, E0.\"Area\" S5, E0.\"CodePublisher\" S6, E0.\"CountryISOCode\" S7, E0.\"CodeID\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (((\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") * ?) > ?) AND (E0.\"ParentDivisionCode\" = ?))\r\n\tbind => [1000000.0, 1000, BE255]\r\nQuery: DataReadQuery(sql=\"SELECT E0.\"ParentDivisionCode\" S0, E0.\"DivisionCode\" S1, E0.\"ParentCodeID\" S2, E0.\"Population\" S3, E0.\"AlternativeCode\" S4, E0.\"Area\" S5, E0.\"CodePublisher\" S6, E0.\"CountryISOCode\" S7, E0.\"CodeID\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (((\"OLINGO\".\"PopulationDensity\"(E0.\"Area\", E0.\"Population\") * ?) > ?) AND (E0.\"ParentDivisionCode\" = ?))\")"}} ==> expected: <200> but was: <500>

[ERROR]   TestScalarDbFunctions.testFilterOnFunctionNested:110 {"error":{"code":null,"message":"Exception [EclipseLink-4002] (Eclipse Persistence Services - 4.0.3.v202405220658): org.eclipse.persistence.exceptions.DatabaseException\r\nInternal Exception: java.sql.SQLSyntaxErrorException: user lacks privilege or object not found: PopulationDensity in statement [SELECT E0.\"ParentCodeID\" S0, E0.\"CodeID\" S1, E0.\"Population\" S2, E0.\"ParentDivisionCode\" S3, E0.\"DivisionCode\" S4, E0.\"Area\" S5, E0.\"CodePublisher\" S6, E0.\"CountryISOCode\" S7, E0.\"AlternativeCode\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (\"OLINGO\".\"PopulationDensity\"(\"OLINGO\".\"ConvertToQkm\"(E0.\"Area\"), E0.\"Population\") > ?)]\r\nError Code: -5501\r\nCall: SELECT E0.\"ParentCodeID\" S0, E0.\"CodeID\" S1, E0.\"Population\" S2, E0.\"ParentDivisionCode\" S3, E0.\"DivisionCode\" S4, E0.\"Area\" S5, E0.\"CodePublisher\" S6, E0.\"CountryISOCode\" S7, E0.\"AlternativeCode\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (\"OLINGO\".\"PopulationDensity\"(\"OLINGO\".\"ConvertToQkm\"(E0.\"Area\"), E0.\"Population\") > ?)\r\n\tbind => [1000]\r\nQuery: DataReadQuery(sql=\"SELECT E0.\"ParentCodeID\" S0, E0.\"CodeID\" S1, E0.\"Population\" S2, E0.\"ParentDivisionCode\" S3, E0.\"DivisionCode\" S4, E0.\"Area\" S5, E0.\"CodePublisher\" S6, E0.\"CountryISOCode\" S7, E0.\"AlternativeCode\" S8 FROM \"OLINGO\".\"AdministrativeDivision\" E0 WHERE (\"OLINGO\".\"PopulationDensity\"(\"OLINGO\".\"ConvertToQkm\"(E0.\"Area\"), E0.\"Population\") > ?)\")"}} ==> expected: <200> but was: <500>
```

**Comment by Nismath** [comment]
are you able to execute mvn clean install archetype:update-local-catalog  cmd?

**Comment by wog48** [comment]
For the original one:  DO you get the error during a maven build or when you execute the test using your IDE?

For the additional question, please see my response to [issue#348](https://github.com/SAP/olingo-jpa-processor-v4/issues/348)

**Comment by spatanjali** [comment]
@wog48 The error is observed while running the maven build (in a local environment as well as in a clear environment within a k8s pod). The error is observed in both the **main** branch as well as **release/2.x.x**. The same is observed even after executing the maven stages (install) on the archtype module. However I couldn't find the archetype-catalog.xml in my .m2 repository. Can you please specify the path where the file should reside.

This issue was not observed while build the previous versions such v1.x or 2.0.x or 2.1.2.

Here are the Maven logs

**mvn build on archtype project**
```[INFO] --- install:3.1.2:install (default-install) @ odata-jpa-archetype-spring ---
[INFO] Installing xxx\olingo-jpa-processor-v4\jpa-archetype\odata-jpa-archetype-spring\pom.xml to xxx\.m2\repository\com\sap\olingo\odata-jpa-archetype-spring\2.1.3\odata-jpa-archetype-spring-2.1.3.pom
[INFO] Installing xxx\olingo-jpa-processor-v4\jpa-archetype\odata-jpa-archetype-spring\target\odata-jpa-archetype-spring-2.1.3.jar to xxx\.m2\repository\com\sap\olingo\odata-jpa-archetype-spring\2.1.3\odata-jpa-archetype-spring-2.1.3.jar
[INFO] 
[INFO] --- archetype:3.2.1:update-local-catalog (default-update-local-catalog) @ odata-jpa-archetype-spring ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18.098 s
[INFO] Finished at: 2024-07-07T12:00:08+02:00
[INFO] ------------------------------------------------------------------------
```

**maven.plugins:maven-archetype-plugin:3.2.1:update-local-catalog**
````[INFO] Scanning for projects...
[INFO] 
[INFO] -------------< com.sap.olingo:odata-jpa-archetype-spring >--------------
[INFO] Building Archetype - odata-jpa-archetype-spring 2.1.3
[INFO]   from pom.xml
[INFO] --------------------------[ maven-archetype ]---------------------------
[INFO] 
[INFO] --- archetype:3.2.1:update-local-catalog (default-cli) @ odata-jpa-archetype-spring ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.261 s
[INFO] Finished at: 2024-07-07T12:05:13+02:00
[INFO] ------------------------------------------------------------------------
````
**mvn build**
```[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for odata-jpa 2.1.3:
[INFO] 
[INFO] odata-jpa .......................................... SUCCESS [  0.562 s]
[INFO] odata-jpa-annotation ............................... SUCCESS [  4.925 s]
[INFO] odata-jpa-vocabularies ............................. SUCCESS [  4.000 s]
[INFO] odata-jpa-odata-vocabularies ....................... SUCCESS [  8.045 s]
[INFO] odata-jpa-test ..................................... SUCCESS [ 13.848 s]
[INFO] odata-jpa-metadata ................................. SUCCESS [ 32.941 s]
[INFO] odata-jpa-processor-ext ............................ SUCCESS [  5.103 s]
[INFO] odata-jpa-processor-cb ............................. SUCCESS [ 25.788 s]
[INFO] odata-jpa-processor ................................ FAILURE [01:07 min]
[INFO] odata-jpa-processor-parallel ....................... SKIPPED
[INFO] odata-jpa-coverage ................................. SKIPPED
[INFO] odata-jpa-spring-support ........................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:42 min
[INFO] Finished at: 2024-07-07T11:58:27+02:00
[INFO] ------------------------------------------------------------------------
```

**Comment by spatanjali** [comment]
> mvn clean install archetype:update-local-catalog

@Nismath  Able to run the above command within the context of the module odata-jpa-archetype

````
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for odata-jpa-archetype 2.1.3:
[INFO] 
[INFO] odata-jpa-archetype ................................ SUCCESS [  0.972 s]
[INFO] Archetype - odata-jpa-archetype-spring ............. SUCCESS [ 17.472 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18.833 s
[INFO] Finished at: 2024-07-07T21:37:09+02:00
[INFO] ------------------------------------------------------------------------
````

**Comment by Nismath** [comment]
@spatanjali is it run with main branch or any another previous release?

**Comment by spatanjali** [comment]
@Nismath It is has been tried on both the _main_ branch as well as _release/2.x.x_

**Comment by Nismath** [comment]
@spatanjali Thank you for the update

**Comment by wog48** [comment]
Actually, I had this problem in one of my workspaces as well. It only occurred when build the project with maven in my IDE. Running the tests via the IDE worked well. The problem did not occur in the pipeline.  I was not able to find the root cause. Even more the problem is gone with the latest changes now available in 2.2.0-SNAPSHOT

**Comment by spatanjali** [comment]
Able to run the maven test and install stages on the Odata-jpa module successfully via the _main_ branch for version 2.2.0-SNAPSHOT

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for odata-jpa 2.2.0-SNAPSHOT:
[INFO] 
[INFO] odata-jpa .......................................... SUCCESS [  0.340 s]
[INFO] odata-jpa-annotation ............................... SUCCESS [  5.873 s]
[INFO] odata-jpa-vocabularies ............................. SUCCESS [  3.916 s]
[INFO] odata-jpa-odata-vocabularies ....................... SUCCESS [  7.359 s]
[INFO] odata-jpa-test ..................................... SUCCESS [ 13.303 s]
[INFO] odata-jpa-metadata ................................. SUCCESS [ 39.129 s]
[INFO] odata-jpa-processor-ext ............................ SUCCESS [  5.209 s]
[INFO] odata-jpa-processor-cb ............................. SUCCESS [ 31.975 s]
[INFO] odata-jpa-processor ................................ SUCCESS [01:47 min]
[INFO] odata-jpa-processor-parallel ....................... SUCCESS [ 12.757 s]
[INFO] odata-jpa-coverage ................................. SUCCESS [  0.093 s]
[INFO] odata-jpa-spring-support ........................... SUCCESS [  0.032 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:47 min
[INFO] Finished at: 2024-08-07T10:33:11+02:00
[INFO] ------------------------------------------------------------------------
```

**Comment by wog48** [comment]
The issue shall be fixed with 2.2.0
